src/mpid/ch4/src/init_comm.c

```diff
@@ -46,6 +46,8 @@ int MPIDI_create_init_comm(MPIR_Comm ** comm)
         }
         mpi_errno = MPIDIG_init_comm(init_comm);
         MPIR_ERR_CHECK(mpi_errno);
+        mpi_errno = MPIDI_NM_mpi_comm_commit_pre_hook(init_comm);
+        MPIR_ERR_CHECK(mpi_errno);
```

Comment:
Add a comment that this is a hacky fix. We probably want to later add `MPIDI_{NM,SHM}_init_comm_hook` so the netmod and shm can properly initialize the init comm -- for example, eliminate potential runtime features for more stability during init.

Reply:
done